crates/core/src/node/mod.rs

```diff
@@ -1253,26 +1255,14 @@ impl PeerId {
     }
 }
 
-thread_local! {
```

Comment:
the reason this exists is because for many tests where we dont care about the peerid (e.g. no integration or e2e tests) generating a large amount of peers is dramatic (comptutationally expensive). For the cases we care about we use the explicit PeerId::random

Arbitrary is used for the most part for unit tests. Be careful with the impact of this change.